vochain/transaction/account_tx.go

```diff
@@ -239,3 +240,133 @@ func (t *TransactionHandler) SetAccountInfoTxCheck(vtx *vochaintx.Tx) error {
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+
+	if _, err := t.state.GetAccount(txAddress, false); err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot get tx account: %w", err)
+	}
+
+	newSik := vtx.Tx.GetSetSik().GetSik()
+	if newSik == nil {
+		return common.Address{}, nil, fmt.Errorf("no sik value provided")
+	}
+	// check if the address already has invalidated sik to ensure that it is
+	// not updated after reach the correct height to avoid double voting
+	if currentSik, err := t.state.SIKFromAddress(txAddress); err == nil {
+		maxEndBlock, err := t.state.ProcessBlockRegistry.MaxEndBlock(t.state.CurrentHeight(), false)
+		if err != nil {
+			return common.Address{}, nil, err
+		}
+		if height := currentSik.DecodeInvalidatedHeight(); height >= maxEndBlock {
+			return txAddress, nil, fmt.Errorf("the sik could not be changed yet")
+		}
+	}
+	return txAddress, newSik, nil
+}
+
+// RegisterSikTxCheck checks if the provided RegisterSikTx is valid ensuring
+// that the proof included on it is valid for the address of the transaction
+// signer.
+func (t *TransactionHandler) RegisterSikTxCheck(vtx *vochaintx.Tx) (common.Address, vstate.SIK, error) {
+	if vtx == nil || vtx.Signature == nil || vtx.SignedBody == nil || vtx.Tx == nil {
+		return common.Address{}, nil, ErrNilTx
+	}
+	tx := vtx.Tx.GetRegisterSik()
+	if tx == nil {
+		return common.Address{}, nil, fmt.Errorf("invalid transaction")
+	}
+	pubKey, err := ethereum.PubKeyFromSignature(vtx.SignedBody, vtx.Signature)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract public key from vtx.Signature: %w", err)
+	}
+	txAddress, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+	if _, err := t.state.GetAccount(txAddress, false); err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot get the account for the tx address: %w", err)
+	}
+	newSik := tx.GetSik()
+	if newSik == nil {
+		return common.Address{}, nil, fmt.Errorf("no sik value provided")
+	}
+	// check if the address already has invalidated sik to ensure that it is
+	// not updated after reach the correct height to avoid double voting
+	if currentSik, err := t.state.SIKFromAddress(txAddress); err == nil {
+		maxEndBlock, err := t.state.ProcessBlockRegistry.MaxEndBlock(t.state.CurrentHeight(), false)
+		if err != nil {
+			return common.Address{}, nil, err
+		}
+		if height := currentSik.DecodeInvalidatedHeight(); height >= maxEndBlock {
+			return common.Address{}, nil, fmt.Errorf("the sik could not be changed yet")
+		}
+	}
+	// check if the address already has a valid sik
+	if _, err := t.state.SIKFromAddress(txAddress); err == nil {
+		return txAddress, nil, fmt.Errorf("this address already has a valid sik")
+	}
+	// get census proof and election information provided
```

Comment:
I'd do all the gets and basic checks (such as the electionID field exist) at the beginning of this function. Since all of these are cheap checks.